File: php/EasyCrypt.php
<?php
/**
 * Created by JetBrains PhpStorm.
 * User: ferron
 * Date: 1/3/12
 * Time: 10:36 PM
 * To change this template use File | Settings | File Templates.
 */
class EasyCrypt
{

    /*
     * CopyLeft (c) 2009 Ferron Hanse  All  Rights Reserved.
     *
     * This software is made available under the terms of the "GPL License"
     * A copy of this license is included with this source distribution
     * in "license.txt" and is also available at:
     * http://www.gnu.org/copyleft/gpl.html
     */

    /*
     * Last Update : 07 September 2009
     *
     */

     /*
      * Welcome to any suggestion : ferronrsmith@gmail.com
      */
     protected $shift = 3;
     protected $cipher = "@)*%/~^!?z";


     public function encrypt($data)
     {
    	$encrypted="";

         for($i=0;$i<strlen($data);$i++)
    		{
                //convert to ascii
    			$compared= ord($data{$i})+$this->shift;
                if($compared<100)
    			{
    				if($compared<10)
    					$encrypted .= "00" . strval($compared);
    				else
    					$encrypted .= "0" . strval($compared);
    			}
    			else {
                    $encrypted .= strval($compared);
                }
            }
    		//index for String substitution
    		$temp ="";
    		for ($j=0;$j<strlen($encrypted);$j++)//substitution
    		{
    			switch($encrypted{$j})
    			{
    				case '0':$temp .= $this->cipher{0};break;
    				case '1':$temp .= $this->cipher{1};break;
    				case '2':$temp .= $this->cipher{2};break;
    				case '3':$temp .= $this->cipher{3};break;
    				case '4':$temp .= $this->cipher{4};break;
    				case '5':$temp .= $this->cipher{5};break;
    				case '6':$temp .= $this->cipher{6};break;
    				case '7':$temp .= $this->cipher{7};break;
    				case '8':$temp .= $this->cipher{8};break;
    				case '9':$temp .= $this->cipher{9};break;
    				default:return;
    			}
    		}
    		$encrypted=$temp;//point $temp address to encrypted
            return $encrypted;
    	}

    	public function decrypt($encrypted)
    	{
    		$encryptedEx="";
    		for($j=0;$j<strlen($encrypted);$j++)//substitution
    		{
    			if($encrypted{$j}==$this->cipher{0})$encryptedEx .= '0';
    			else if($encrypted{$j}==$this->cipher{1})$encryptedEx .= '1';
    			else if($encrypted{$j}==$this->cipher{2})$encryptedEx .= '2';
    			else if($encrypted{$j}==$this->cipher{3})$encryptedEx .= '3';
    			else if($encrypted{$j}==$this->cipher{4})$encryptedEx .= '4';
    			else if($encrypted{$j}==$this->cipher{5})$encryptedEx .= '5';
    			else if($encrypted{$j}==$this->cipher{6})$encryptedEx .= '6';
    			else if($encrypted{$j}==$this->cipher{7})$encryptedEx .= '7';
    			else if($encrypted{$j}==$this->cipher{8})$encryptedEx .= '8';
    			else $encryptedEx .='9';
    		}

    		$encrypted=$encryptedEx;
    		$data="";
    		$temp=0;
    		for($i=0;$i<strlen($encrypted);$i=$i+3)
    		{
    			$temp= $encrypted{$i} . "" . $encrypted{$i+1} . "" . $encrypted{$i+2};
                $temp-=$this->shift;
                //convert value to int then from ascii to char
    			$data .= chr(floor($temp));
    		}
    		return $data;
    	}
}


?>

File: easycrypt.py
"""
* Created by JetBrains PhpStorm.
* User: ferron
* Date: 1/3/12
* Time: 5:01 PM
* To change this template use File | Settings | File Templates.
*/

/*
* CopyLeft (c) 2012 Ferron Hanse  All  Rights Reserved.
*
* This software is made available under the terms of the "GPL License"
* A copy of this license is included with this source distribution
* in "license.txt" and is also available at:
* http://www.gnu.org/copyleft/gpl.html
*
* Last Update : 19 February 2012
* Converted to Python, Java, PHP, JavaScript
*
* Welcome to any suggestion : ferronrsmith@gmail.com
*
"""

# Much like a ROT13 shift used in the caesar cipher
# http://en.wikipedia.org/wiki/Caesar_cipher
SHIFT = 3
CIPHER = '@)*%/~^!?z' # cipher : keep this secret ^_^

def encrypt (data) :
    # private variables
    encrypted = "" # default string for storing result
    temp = ""
    j = 0
    for i in range(len(data)):
        compared = ord(data[i]) + SHIFT
        if compared < 100:
            if compared < 10 :
                encrypted += '00' + str(compared)
            else:
                encrypted += '0' + str(compared)
        else:
            encrypted += str(compared)

    for j in range(len(encrypted)):
        if encrypted[j] == '0':
            temp += CIPHER[0]
        elif encrypted[j] == '1':
            temp += CIPHER[1]
        elif encrypted[j] == '2':
           temp += CIPHER[2]
        elif encrypted[j] == '3':
            temp += CIPHER[3]
        elif encrypted[j] == '4':
            temp += CIPHER[4]
        elif encrypted[j] == '5':
            temp += CIPHER[5]
        elif encrypted[j] == '6':
            temp += CIPHER[6]
        elif encrypted[j] == '7':
            temp += CIPHER[7]
        elif encrypted[j] == '8':
            temp += CIPHER[8]
        elif encrypted[j] == '9':
            temp += CIPHER[9]
        else:
            print "Encrypt Error:number out of range"
    encrypted = temp
    return encrypted

def decrypt(data):
    # private variables
    encrypted=""
    result = ""
    temp = 0
    inc = 0
    for j in range(len(data)):
        if data[j] == CIPHER[0]:
            encrypted +='0'
        elif data[j] == CIPHER[1]:
            encrypted += '1'
        elif data[j] == CIPHER[2]:
            encrypted += '2'
        elif data[j] == CIPHER[3]:
            encrypted += '3'
        elif data[j] == CIPHER[3]:
            encrypted += '3'
        elif data[j] == CIPHER[4]:
            encrypted += '4'
        elif data[j] == CIPHER[5]:
            encrypted += '5'
        elif data[j] == CIPHER[6]:
            encrypted += '6'
        elif data[j] == CIPHER[7]:
            encrypted += '7'
        elif data[j] == CIPHER[8]:
            encrypted += '8'
        else:
            encrypted += '9'
    data = encrypted

    while inc < len(data):
        temp = int(data[inc] + data[inc+1] + data[inc+2])
        temp -= SHIFT
        result += chr(temp)
        inc += 3

    return result